src/go/k8s/cmd/configurator/main.go

```diff
@@ -159,20 +179,129 @@ func main() {
-	cfg *config.Config, c configuratorConfig, index brokerID,
+	cfg *config.Config, c configuratorConfig, index brokerID, k8sClient client.Client,
 ) error {
+	if index != 0 {
+		return nil
+	}
+
 	empty, err := IsRedpandaDataFolderEmpty(c.dataDirPath)
 	if err != nil {
 		return fmt.Errorf("checking Redpanda data folder content (%s): %w", c.dataDirPath, err)
 	}
 
-	if index == 0 && empty {
+	if !empty {
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	stsName := renderStsName(c.hostName, index)
+
+	sts := &v1.StatefulSet{}
+	err = k8sClient.Get(ctx, client.ObjectKey{
+		Name: stsName,
+	}, sts)
+	if err != nil {
+		return fmt.Errorf("retrieving statful set: %w", err)
+	}
+
+	if *sts.Spec.Replicas == 1 {
 		cfg.Redpanda.SeedServers = []config.SeedServer{}
+		return nil
+	}
+
+	podList := &corev1.PodList{}
+	err = k8sClient.List(ctx, podList, &client.ListOptions{
+		LabelSelector: k8sLabels.SelectorFromSet(map[string]string{
+			labels.InstanceKey: stsName,
+		}),
+	})
+	if err != nil {
+		return fmt.Errorf("listing available PODs: %w", err)
+	}
+
+	anyRedpandaPodIsRunning := false
+	for _, pod := range podList.Items {
+		if pod.Name == c.hostName {
+			continue
+		}
+		if pod.Status.Phase == corev1.PodRunning {
+			anyRedpandaPodIsRunning = true
+			break
+		}
 	}
 
+	if anyRedpandaPodIsRunning {
+		for _, srv := range cfg.Redpanda.SeedServers {
+			cl := &redpandav1alpha1.Cluster{}
+			err = k8sClient.Get(ctx, client.ObjectKey{
+				Name: stsName,
+			}, cl)
+			if err != nil {
+				return fmt.Errorf("retrieving cluster object: %w", err)
+			}
+
+			scheme := "http"
+			if len(cl.Spec.Configuration.AdminAPI) > 0 && cl.Spec.Configuration.AdminAPI[0].TLS.Enabled {
+				scheme = "https"
+			}
+			// TODO: Implement AdminAPI that has RequireClientAuth set to true (mTLS)
+			request, err := http.NewRequestWithContext(ctx, http.MethodGet, fmt.Sprintf("%s://%s:%d/v1/brokers", scheme, srv.Host.Address, srv.Host.Port), nil)
+			if err != nil {
+				return fmt.Errorf("creating seed server broker reques: %w", err)
+			}
+
+			tr := http.DefaultTransport.(*http.Transport).Clone()
+			// TODO: construct client with root CA for AdminApi that should be mounted
+			// to the configurator container.
+			tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+			httpClient := &http.Client{Transport: tr}
+			resp, err := httpClient.Do(request)
+			if err != nil {
+				// init container need to be robust in retrieving other brokers cluster view
+				// any error might block Redpanda initialization
+				continue
+			}
+			b, err := io.ReadAll(resp.Body)
+			resp.Body.Close()
+			if err != nil {
+				return fmt.Errorf("reading seed server broker body: %w", err)
+			}
+
+			var brokers []broker
+			err = json.Unmarshal(b, &brokers)
+			if err != nil {
+				return fmt.Errorf("unmarshalling brokers response: %w", err)
+			}
+
+			for _, br := range brokers {
+				if br.NodeId == 0 {
```

Comment:
`nodeId` is no longer the `podOrdinal`, so this check may never hold

Reply:
You are right. Fixed in next push.